app/src/main/java/com/csc/telezhnaya/todo2/MainActivity.java

```diff
@@ -0,0 +1,117 @@
+
+    @Bind(R.id.list)
+    ListView listView;
+    @Bind(R.id.new_task)
+    EditText newTask;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.layout_main);
+        ButterKnife.bind(this);
+        taskManager.bind(this);
+        listView.setAdapter(taskManager.getAdapter());
+        getSupportLoaderManager().initLoader(0, null, taskManager);
+    }
+
+    @Override
+    protected void onDestroy() {
+        super.onDestroy();
+        taskManager.unbind();
+    }
+
+    public void onAddClick(View view) {
+        String text = newTask.getText().toString();
+        if (!text.isEmpty()) {
+            newTask.setText("");
+            taskManager.addTask(text);
+        }
+    }
+
+    public void onTaskChanged(View view) {
+        View parent = (View) view.getParent();
+        TextView taskName = (TextView) parent.findViewById(R.id.task);
+        CheckBox checkBox = (CheckBox) parent.findViewById(R.id.task_done);
+        ToggleButton star = (ToggleButton) parent.findViewById(R.id.star);
+        taskManager.updateTask((int) parent.getTag(), taskName.getText().toString(), null, checkBox.isChecked(), star.isChecked());
+    }
+
+    public void onTextClick(View view) {
+        View parent = (View) view.getParent();
+        final int position = (int) parent.getTag();
+        TextView task = (TextView) parent.findViewById(R.id.task);
+        CheckBox checkBox = (CheckBox) parent.findViewById(R.id.task_done);
+        ToggleButton toggleButton = (ToggleButton) parent.findViewById(R.id.star);
+        TextView d = (TextView) parent.findViewById(R.id.creation_date);
+
+        AlertDialog.Builder dialogBuilder = new AlertDialog.Builder(this);
+        final View dialogView = this.getLayoutInflater().inflate(R.layout.edit_task, null);
+        dialogBuilder.setView(dialogView);
+        final EditText taskName = (EditText) dialogView.findViewById(R.id.task_name);
+        taskName.setText(task.getText().toString());
+        final EditText taskDescription = (EditText) dialogView.findViewById(R.id.task_description);
+        taskDescription.setText(taskManager.getDescription(position));
+        TextView date = (TextView) dialogView.findViewById(R.id.creation_date);
+        date.setText(d.getText().toString());
+        final CheckBox done = (CheckBox) dialogView.findViewById(R.id.task_done);
+        done.setChecked(checkBox.isChecked());
+        final ToggleButton star = (ToggleButton) dialogView.findViewById(R.id.star);
+        star.setChecked(toggleButton.isChecked());
+
+
+        ListView listView = (ListView) dialogView.findViewById(R.id.tags);
+        final TagManager tagManager = new TagManager(position);
+        tagManager.bind(this);
+        listView.setAdapter(tagManager.getAdapter());
+        getSupportLoaderManager().restartLoader(1, null, tagManager);
```

Comment:
Yes, you are right, I was thinking about it. I need here initLoader with "position" variable as first argument.